app-dropwizard/src/main/java/net/spals/appbuilder/app/dropwizard/DropwizardWebApp.java

```diff
@@ -40,7 +41,11 @@
 
         public Builder(final Bootstrap<?> bootstrap, final Logger logger) {
             this.appDelegateBuilder = new JaxRsWebApp.Builder(bootstrap.getApplication().getName(), logger);
-            setBootstrap(bootstrap);
+            // A little bit of a hack here, since I'm not super keen on constructors
+            // with side-effects. But given the Dropwizard pattern espoused by AppBuilder,
+            // we really need the ConfigurationSourceProvider registered with the Bootstrap
+            // as soon as possible.
+            registerConfigurationSourceProvider(bootstrap);
```

Comment:
Like the comment says, a little bit of a hack since I don't like constructors with side effects. But I think this is the best way to keep things simple from the client side.